content/en/docs/reference/command-line-tools-reference/feature-gates/watch-list-client.md

```diff
@@ -0,0 +1,16 @@
+---
+title: WatchListClient
```

Comment:
The `WatchListClient` is defined for a client (the client-go library) and not for the server.

It enables the client to request a stream of data rather than an entire list. The streamed data is then transformed into a list response on the client side.

Another way to think about this feature is that it essentially replaces standard LIST calls with WATCH calls under the hood. 
Individual list items are delivered via the WATCH call and transformed into a list response on the client. 
This process is completely opaque to the client requesting a list. 
For the client, it doesn’t matter whether the library issues a LIST or WATCH request/verb. 
The client only cares about the result, which will be a list response.

Also, for the `WatchListClient` to work, the server must support API Streaming, which is controlled by a server-side `WatchList` feature flag. If the server doesn't support the API Streaming and the `WatchListClient` is enabled then the client falls back/issues to a standard LIST request.

Reply:
Well this feature gate be configurable on API clients such as: the kubelet, kube-controller-manager, the scheduler?

Reply:
> The WatchListClient is defined for a client (the client-go library) and not for the server.

So if I'm interacting with a Kubernetes cluster using pure RESTful API, and I'm developing
a client tool in Python or Rust, this feature gate doesn't matter at all. But ...
it is listed in the command line options for `kube-apiserver`, `kube-controller-manager`, `kube-scheduler`, `kube-proxy` and `kubelet`. Right?

Reply:
I assume you can configure some of those components to enable or not enable this feature when acting as a client of the API server.

Reply:
> [sftim](https://github.com/sftim) [2 minutes ago](https://github.com/kubernetes/website/pull/47261#discussion_r1692902844)
Well this feature gate be configurable on API clients such as: the kubelet, kube-controller-manager, the scheduler?

yes, kubernetes will expose the `WatchListClient` feature gate via command line options for some/all control plane components. Initially we were planning to enable the `WatchListClient` by default only for `KCM`.